cmd/promlts/sidecar.go

```diff
@@ -93,7 +116,56 @@ func runSidecar(
 			s.Stop()
 			l.Close()
 		})
+	}
+	// Periodically query the Prometheus config. We use this as a heartbeat as well as for updating
+	// the external labels we apply.
+	{
+		promUp := prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "promlts_sidecar_prometheus_up",
+			Help: "Boolean indicator whether the sidecar can reach its Prometheus peer.",
+		})
+		lastHeartbeat := prometheus.NewGauge(prometheus.GaugeOpts{
+			Name: "promlts_sidecar_last_heartbeat_seconds",
+			Help: "Second timestamp of the last attempted heartbeat",
+		})
+		reg.MustRegister(promUp, lastHeartbeat)
+
+		ctx, cancel := context.WithCancel(context.Background())
+		tick := time.NewTicker(30 * time.Second)
+
+		do := make(chan struct{}, 1)
+		do <- struct{}{}
+
+		go func() {
+			for range tick.C {
+				do <- struct{}{}
+			}
+		}()
 
+		g.Add(func() error {
+			for {
+				select {
+				case <-ctx.Done():
+					return nil
+				case <-do:
```

Comment:
Why not just `<-tick.C` here? and remove `do` channel logic?

Reply:
It will take the timer 30 seconds to fire for the first time. If we want things to start working right after startup we've to work around that somehow – it's either this or duplicating the iteration code once before the for loop.

Reply:
well, if we want to not wait for the first time, why we don't put this `select` at the end of this `for` loop?